openedx/core/djangoapps/notifications/base_notification.py

```diff
@@ -465,14 +476,47 @@ def get_notification_content(notification_type, context):
         'strong': 'strong',
         'p': 'p',
     }
+    content_function = get_notification_type_content_function(notification_type)
     notification_type = NotificationTypeManager().notification_types.get(notification_type, None)
     if notification_type:
+        if content_function:
+            return content_function(notification_type, context)
         notification_type_content_template = notification_type.get('content_template', None)
         if notification_type_content_template:
             return notification_type_content_template.format(**context, **html_tags_context)
     return ''
 
 
+def get_new_comment_notification_content(notification_type, context):
+    """
+        Returns notification content for the new_comment notification.
+    """
+    return get_notification_content_with_author_pronoun(notification_type, context)
+
+
+def get_comment_on_followed_post_notification_content(notification_type, context):
+    """
+        Returns notification content for the comment_on_followed_post notification.
+    """
+    return get_notification_content_with_author_pronoun(notification_type, context)
+
+
+def get_notification_content_with_author_pronoun(notification_type, context):
+    """
+        Helper function to get notification content with author's pronoun.
+    """
+    html_tags_context = {
+        'strong': 'strong',
+        'p': 'p',
+    }
+    notification_type_content_template = notification_type.get('content_template', None)
+    if 'author_pronoun' in context:
+        context['author_name'] = context['author_pronoun']
+    if notification_type_content_template:
+        return notification_type_content_template.format(**context, **html_tags_context)
+    return ''
+
```

Comment:
Nit: Best practice when we have two same functions `func_a` and `func_b` is to create `func_a` and then assign `func_b  = func_a`

```
def get_notification_content_with_author_pronoun(notification_type, context):
    ...
    return ''

get_new_comment_notification_content = get_notification_content_with_author_pronoun

get_comment_on_followed_post_notification_content = get_notification_content_with_author_pronoun
```
OR
```
def get_new_comment_notification_content(notification_type, context):
    """
        Helper function to get notification content for new_comment notification type with author's pronoun.
    """
    html_tags_context = {
        'strong': 'strong',
        'p': 'p',
    }
    notification_type_content_template = notification_type.get('content_template', None)
    if 'author_pronoun' in context:
        context['author_name'] = context['author_pronoun']
    if notification_type_content_template:
        return notification_type_content_template.format(**context, **html_tags_context)
    return ''


get_comment_on_followed_post_notification_content = get_new_comment_notification_content
```